.github/workflows/pr_tests.yml

```diff
@@ -0,0 +1,64 @@
+name: Full CI Platform Tests
+# This workflow runs on pull requests to test the code across multiple platforms and architectures.
+
+on:
+  pull_request:
+    # Trigger the workflow on PR events, specifically when opened or updated.
+    paths:
+      - '**/*.rs'  # Run only if there are changes in Rust source files.
```

Comment:
Good stuff

Reply:
Yes - so the CI workflow is basically that `ci.yml` runs on all push/commit/PR.
Then `pr_tests` only runs on PRs (not every commit - commit tests are quick as they use the built in runner and lint checks).
Then if `Full Tests` pass AND the PR is tagged with a vnum, `release_macos.yaml` runs, auto builds, and pushes the release to the releases/assets of the repository.